### PR TITLE
Created new endpoint GetContactInformationByEmailAddress 

### PR DESCRIPTION
**Pull request by suhasini3091**
## Description
Created new endpoint GetContactInformationByEmailAddress that returns contactInformations of organizations where the given emailAddress is registered
Added a new index to the userContactInfo table email address field 
The call is now authenticated using the Maskinporten scope

**Enhancements**
  * Contact responses include Organization Number.
  * National identity number now optional; users without SSN are returned.
  * Relaxed profile filtering to include users when name is present.

## Related Issue(s)
- #589 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Summary by CodeRabbit

* **New Features**
  * Email-based lookup for professional contact information via the dashboard API
  * Contact responses now include organization number; national identity number may be null

* **Bug Fixes / Behavior**
  * Organization lookups return 200 OK with an empty list when no contacts found (instead of 404)
  * Identity enrichment returns entries even when SSN is missing

* **Tests**
  * Expanded coverage for email- and organization-based lookups, edge cases, and error paths

* **Chores**
  * Added a database index on email address for improved query performance
* **Documentation**
  * Updated README and added API test docs for email endpoint

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/Altinn.Profile.Integrations/Migration/v0.21/01-add-index-for-user-party-contact-info.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+CREATE INDEX IF NOT EXISTS ix_user_party_contact_info_email_address ON professional_notification_settings.user_party_contact_info (email_address);
```
[the remark this answers is not in the document]
Comment (suhasini3091) [reply src/Altinn.Profile.Integrations/Migration/v0.21/01-add-index-for-user-party-contact-info.sql]: Yes I have confirmed that there is no preexisting index on email address, there is no need for table name within the index

[the remark this answers is not in the document]
Comment (sigridbra) [reply src/Altinn.Profile.Integrations/Migration/v0.21/01-add-index-for-user-party-contact-info.sql]: Please make sure that the dbContext and the migration uses the same index name

File: src/Altinn.Profile/Controllers/DashboardController.cs [hunk 2]
```diff
@@ -216,5 +214,51 @@ public async Task<ActionResult<List<DashboardUserContactInformationResponse>>> G
...
+                return ValidationProblem(ModelState);
+            }
+
+            var contactInfosByEmail = await _professionalNotificationsService
+                .GetContactInformationByEmailAddressAsync(emailAddress, cancellationToken);
+
+            if (contactInfosByEmail.Count == 0)
+            {
+                return NotFound();
+            }
```
Comment (sigridbra) [inline src/Altinn.Profile/Controllers/DashboardController.cs]: Actually the task specifies so we should rather return 200 with empty list. The same goes for the other endpint in this controller 

>  Empty or missing organisation results in a successful response with an empty collection and no partial failures.
